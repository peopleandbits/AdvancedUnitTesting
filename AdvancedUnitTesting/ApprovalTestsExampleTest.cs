using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using Newtonsoft.Json;
using ApprovalTests.Reporters;
using AdvancedUnitTesting.Domain;

namespace AdvancedUnitTesting
{
    [TestClass]
    public class ApprovalTestsExampleTest
    {
        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void ApprovalsComplexObject()
        {
            // assign
            var p = Student.Default();

            // act

            // assert
            Approvals.Verify(p);
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void ApprovalsComplexObjectJson()
        {
            // assign
            var p = Student.Default();

            // act

            // assert
            Approvals.VerifyJson(JsonConvert.SerializeObject(p));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void ApprovalsArray()
        {
            // assign
            var p = Student.Default();

            // act

            // assert... old way
            string pSerialized = JsonConvert.SerializeObject(p, Formatting.Indented);
            Approvals.VerifyAll(p.SchoolHistory.Select(c=>c.Name), "Schlool");
        }
    }
}