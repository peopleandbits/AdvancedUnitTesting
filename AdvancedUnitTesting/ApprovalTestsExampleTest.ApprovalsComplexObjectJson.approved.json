{
    "Age": 30,
    "Name": "Tommy",
    "SchoolHistory": [
        {
            "Name": "Freising Grundschule"
        },
        {
            "Name": "Freising Gymnasium"
        },
        {
            "Name": "Melbourne Hacker Academy"
        },
        {
            "Name": "Lissabon University for Star Wars Fans"
        }
    ]
}